website/website/website.py

```diff
@@ -0,0 +1,120 @@
+from typing import Set
+import os
+from aiohttp import web
+import jinja2
+import logging
+import aiohttp_session
+import glob
+
+from hailtop.config import get_deploy_config
+from hailtop.tls import internal_server_ssl_context
+from hailtop.hail_logging import AccessLogger
+from gear import setup_aiohttp_session, web_maybe_authenticated_user
+from web_common import (setup_aiohttp_jinja2, setup_common_static_routes, render_template,
+                        sass_compile)
+
+
+MODULE_PATH = os.path.dirname(__file__)
+log = logging.getLogger('scorecard')
+deploy_config = get_deploy_config()
+routes = web.RouteTableDef()
+
+
+def redirect(from_url, to_url):
+    async def serve(request):  # pylint: disable=unused-argument
+        raise web.HTTPFound(to_url)
+    routes.get(from_url)(serve)
+
+
+@routes.get('/healthcheck')
+async def get_healthcheck(request):  # pylint: disable=unused-argument
+    return web.Response()
+
+
+@routes.get('/robots.txt')
+async def get_robots(request):  # pylint: disable=unused-argument
+    return web.Response('user-agent: *\nAllow: /')
+
+
+routes.static('/static', f'{MODULE_PATH}/static')
+redirect('/favicon.ico', 'common_static/hail_logo_sq.ico')
+redirect('/docs/batch', 'batch/index.html')
+redirect('/docs/batch/', 'index.html')
+redirect('/docs/0.2', '0.2/index.html')
+redirect('/docs/0.2/', 'index.html')
+
+
+DOCS_PATH = f'{MODULE_PATH}/docs/'
+STATIC_DOCS_PATHS = ['0.2/_static',
+                     '0.2/_sources',
+                     '0.2/tutorials',
+                     'batch/_static',
+                     'batch/_images',
+                     'batch/_sources']
+FQ_STATIC_DOCS_PATHS: Set[str] = set()
+
+
+for path in STATIC_DOCS_PATHS:
+    routes.static('/docs/' + path, DOCS_PATH + path)
+    FQ_STATIC_DOCS_PATHS.add(DOCS_PATH + path)
+
+
+docs_pages = set(
+    x[0][len(DOCS_PATH):] + '/' + y
+    for x in os.walk(DOCS_PATH)
+    if x[0] not in FQ_STATIC_DOCS_PATHS
+    for y in x[2])
```

Comment:
This reads OK, but honestly, it still seems brittle.  There's no test for the `x[0] not in FQ_STATIC_DOCS_PATHS` condition being right, for example.  I feel like the main problem here is that we're mixing static and templated files  in docs/, but not elsewhere.  Is there a reason we have to do that?

Reply:
Sphinx is generating the structure. I'd have to walk all the HTML and replace URLs that reference `_static`. If we had our own doc generator this wouldn't be an issue.